### PR TITLE
Replace maiigned by fieldalignment

### LINKED ISSUE #987: Switch maligned linter to govet fieldalignment
Since the `maiigned` linter is deprecated we should change to `govet 'fieldalignment' `.

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancasa@gmail.com>

Fixes #987

### CODE REVIEW

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml [hunk 1]
```diff
@@ -249,10 +247,8 @@ spec:
                   attached to Collector and Target Allocator pods.
                 type: object
               podSecurityContext:
-                description: PodSecurityContext holds pod-level security attributes
-                  and common container settings. Some fields are also present in container.securityContext.  Field
-                  values of container.securityContext take precedence over field values
-                  of PodSecurityContext.
+                description: SecurityContext will be set as the container security
```
Comment (yuriolisa) [inline config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:250]: Which version did you use to create the bundle?

**Review by yuriolisa — COMMENTED** [review]
Could you please ensure you have submitted the values related to the current operator-sdk version?

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml (hunk 1 above)
Comment (iblancasa) [reply config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:250]: Oh sorry! I found the problem (I though the change was expected but not). I used the `fix` option from `fieldalignment` and it removes the descriptions. I fixed some of them manually but it seems I broke others. Working on the fix.

Thanks for the feedback and sorry for the inconvenience :)

**Review by yuriolisa — APPROVED** [review]
LGTM.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**